doc/ssh_config.md

```diff
@@ -0,0 +1,230 @@
+
+###  1.7. <a name='High-LevelDesign'></a>High-Level Design 
+
+We want to enable global ssh server configuration in SONIC. In order to do so will touch few areas in the system:
+1. configDB - to include a dedicated table for configurations
+2. hostcfg demon - to update ssh config files once configDB relevant areas are modified (and for this feature, ssh global config table)
+3. OS ssh config files - specific for this stage we are only /etc/ssh/sshd_config is going to be modifed by the hostcfg demon.
+4. OS ssh service - to be restarted after each configuration change.
+
+Note:
+
+The Daemon is running in the host (without container) that matches this feature, because it is basically writing policies on ssh config files from the OS.
+##### Flow diagram
+![ssh_flow_community](ssh_flow_community.png)
+#### 1.7.1 <a name='Flow description'></a>Flow description
+When the feature is enabled, by modifying the DB manually, user will set ssh server policies/configuration (see options below) by modifing CONF_DB in SSH_GLOBAL_TABLE.
+
+The hostcfgd daemon will be extended to listen to ssh policies/configurations from SSH_GLOBAL table, parse the inputs and set the new policies to ssh config files, and update ssh server afterwards.
+
+
+#### 1.7.2 <a name='SSH global policies'></a>SSH global policies
+
+We want to enable configuring the following policies, with default values are taken from OS (Debian):
+| Policy         |      Action                                                               | Param values            | Default OS value |
+|----------------|---------------------------------------------------------------------------|-------------------------|------------------|
+| login attempts |     Number of attempts to try to log in   before rejecting the session    |     3-100               |     6            |
+| login timeout  |     SSH session timeout                                                   |     1-600 (secs)        |     120          |
+| ports          |     Port number for SSH                                                   |     1-65535             |     22           |
+| tcp-forwarding |     enable the option of tcp forwarding for   the ssh server              |     enabled/disabled    |     enabled      |
+| x11-forwarding |     enable the option of x11 forwarding for   the ssh server              |     enabled/disabled    |     enabled      |
+
+###  1.8. <a name='Init flow'></a>Init flow 
+
+During init flow we will set default ssh policies, same as default values in DebianOS. Default values will be added to init_cfg.json.j2, and updated in sshd_config file accordingly.
+####  1.8.1. <a name='FeatureDefault'></a>Feature Default
+
+Description of default values in init_cfg.json regarding SSH global config:
+```
+login attempts: 6 
+login timeout: 120 //seconds
+ports: 22
+tcp-fowarding: Enabled

```

Comment:
This should be 'true' as per config_db schema. Please modify

Reply:
done

Reply:
irrelevant for now